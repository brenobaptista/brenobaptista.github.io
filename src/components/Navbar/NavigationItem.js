import React from 'react'
import { Link } from 'gatsby'
import PropTypes from 'prop-types'

const NavigationItem = ({ link, clicked, children }) => (
  <li className='navigationItem'>
    <Link
      to={link}
      activeClassName='active'
      onClick={clicked}
    >
      {children}
    </Link>
  </li>
)

NavigationItem.propTypes = {
  link: PropTypes.string,
  clicked: PropTypes.func,
  children: PropTypes.node,
}

NavigationItem.defaultProps = {
  link: '',
  clicked: () => {},
  children: '',
}

export default NavigationItem
