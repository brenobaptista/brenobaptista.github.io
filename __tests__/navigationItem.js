import React from 'react'
import { render } from '@testing-library/react'
import NavigationItem from '../src/components/Navbar/NavigationItem'

test('matches snapshot', () => {
  const clicked = jest.fn()
  const { asFragment } = render(<NavigationItem link='' clicked={clicked} />)

  expect(asFragment(<NavigationItem link='' clicked={clicked} />)).toMatchSnapshot()
})
