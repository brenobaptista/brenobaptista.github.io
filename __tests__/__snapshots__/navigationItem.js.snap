// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="navigationItem"
  >
    <a
      href=""
    />
  </li>
</DocumentFragment>
`;
